app/src/main/java/com/alphawallet/app/walletconnect/TransactionDialogBuilder.java

```diff
@@ -205,20 +207,6 @@ private void reject(Throwable error, AWWalletConnectClient awWalletConnectClient
     private void approve(String hashData, AWWalletConnectClient awWalletConnectClient)
     {
         new Handler(Looper.getMainLooper()).postDelayed(() ->
-                awWalletConnectClient.approve(sessionRequest, hashData), 5000);
-    }
-
-    @Override
-    public void onCancel(@NonNull DialogInterface dialog)
```

Comment:
Just a reminder, `onCancel` and `onDismiss` were introduced in https://github.com/AlphaWallet/alpha-wallet-android/issues/3040

Reply:
> Just a reminder, `onCancel` and `onDismiss` were introduced in #3040

Ah ok - well spotted - it was causing an issue where when the sheet was dismissed it would always return 'fail', even if the transaction was successful. Maybe we can pass on the fail message from the inner sheet?

Reply:
Reinstated and fixed so it doesn't interfere with the successful signing.